### ISSUE TITLE
make npm module

### ISSUE THREAD

**Issue by matthiasg** [mozilla-b2g/jsas#23]
any plans on making this a full library with node.js support ?

**Comment by asutherland** [comment]
tl;dr: no plans, but not opposed if someone else wants to take the library in that direction as long as web compatibility is maintained.

Not sure what you mean by full library.  Our only known consumer at this point is the Firefox OS Gaia email app that we created this lib for that runs in a RequireJS-family AMD module loader space in the browser.  The Gaia email app's backend provides higher level functionality and so the primary goal of this library is to be a protocol layer, roughly akin to https://github.com/whiteout-io/browserbox is for IMAP.  You can see our use of the library at https://github.com/mozilla-b2g/gaia-email-libs-and-more/tree/master/js/activesync.  It's quite possible that some of the logic in there may instead belong in jsas.

I don't think anyone's opposed to making the library more friendly to node.js; certainly the intent of the UMD-wrapper was to make it feasible to use it in node.js.  (Our worker-thread branch is a little hacked up out of necessity and is probably not the right branch for that.)  We're not particularly fans of ActiveSync or the many broken server implementations out there, and we are trying to use IMAP whenever possible, so the Mozilla Corporation engineers working on the email app are unlikely to move this forward.  But I think we'd be happy to accept patches into this repo, and/or if you or others out there really want to make an impressive JS ActiveSync library and our bandwidth is insufficent, we'd be happy to just be a (downstream) consumer of a thriving upstream.
